comm/server.go

```diff
@@ -99,26 +116,79 @@ func HandleSighup() {
 	// Reload Accounts and Jobs
 	fmt.Println("Reloading Accounts")
 
-	server := server()
+	server := Global()
 
-	server.Auth.Lock()
+	// Reload accounts
 	server.Auth.Reload()
-	server.Auth.Unlock()
 
-	log.Dump("Accounts", server.Auth.Map)
-
-	// TODO: SIGHUP seems to hang the server, not sure if this is
+	// TODO: SIGHUP seems to hang the web server, not sure if this is
 	// a reasonable way to restart it, or it's just leaking ...
-	err := server.web.ListenAndServe()
-	if err != nil {
-		fmt.Println("FATAL: Web Server Error: ", err)
-		return
-	}
+	// Probably need a better way to tell the server to reload
+	fmt.Println("Restarting Web Server after Signal")
+	TryWebServerRestart()
```

Comment:
I really wouldn't worry about any of this signal based reloading, it feels really like scope creep on what's just meant to be a prototype. There is nothing wrong with just having to restart a process wen something external changes.

Reply:
I had figured since I should handle SIGCHLD for reaping anyways, that SIGINT for reloading would be fine. I was surprised to find out that was interfering with net/http, then again later with fsnotify. If I had time I would have dug into what that was happening (since I find it kinda scary).